### DISCUSSION TITLE
"object has no attribute" error

### DISCUSSION THREAD

**Discussion by tompayne36** [jamesoff/simplemonitor#665] [Q&A]
Hi,

I am trying to get simplemonitor to work on my Raspberry Pi but am getting an "object has no attribute" error. I have tried to do this on an Ubuntu image on AWS EC2 but get the identical error. I believe I am using the latest "git clone" and have installed the "arrow" package (my first error). 

I am new to python on linux (and github), so apologize if this is a noob question.

========

ubuntu@ip-172-31-53-23:~$ cd simplemonitor/
ubuntu@ip-172-31-53-23:~/simplemonitor$ python3 monitor.py
Traceback (most recent call last):
  File "monitor.py", line 1, in <module>
    from simplemonitor import monitor
  File "/home/ubuntu/simplemonitor/simplemonitor/monitor.py", line 10, in <module>
    from .simplemonitor import SimpleMonitor
  File "/home/ubuntu/simplemonitor/simplemonitor/simplemonitor.py", line 15, in <module>
    from .Alerters.alerter import Alerter
  File "/home/ubuntu/simplemonitor/simplemonitor/Alerters/__init__.py", line 5, in <module>
    from .bulksms import BulkSMSAlerter
  File "/home/ubuntu/simplemonitor/simplemonitor/Alerters/bulksms.py", line 7, in <module>
    from ..Monitors.monitor import Monitor
  File "/home/ubuntu/simplemonitor/simplemonitor/Monitors/__init__.py", line 23, in <module>
    from .service import (
  File "/home/ubuntu/simplemonitor/simplemonitor/Monitors/service.py", line 346, in <module>
    class MonitorProcess(Monitor):
  File "/home/ubuntu/simplemonitor/simplemonitor/Monitors/service.py", line 376, in MonitorProcess
    ) -> List[psutil.Process]:
AttributeError: 'NoneType' object has no attribute 'Process'
ubuntu@ip-172-31-53-23:~/simplemonitor$

**Comment by jamesoff** [comment]
Hi! Sorry you've having problems. This is actually useful, as I've seen that error intermittently myself before but not managed to reproduce it to debug it. Would you be able to share your `monitors.ini`? At least any "process" monitors you have (as that's the one the error is occurring in).

Edit: Looking at the code, I think this is actually a different error to my intermittent one; I think you're missing the `psutil` package too. You probably need to do `pip install .` in the directory you've got simplemonitor cloned into, which should pull in all the packages it depends on.

**Reply by tompayne36** [reply]
Thanks..."pip install ." is new to me. I am making more progress now after installing psutil. I am now working through likely issues in my monitor.ini and monitors.ini files, but that is on me.

**Reply by tompayne36** [reply]
Making progress...now only getting this warning: 

/usr/lib/python3/dist-packages/requests/__init__.py:91: RequestsDependencyWarning: urllib3 (1.26.2) or chardet (3.0.4) doesn't match a supported version!
  RequestsDependencyWarning)

But I assume I can ignore this for now...

**Reply by jamesoff** [reply]
If it's working then yes :)

Might be worth creating a virtualenv (in your simplemonitor git dir: `python3 -m venv .venv; source .venv/bin/activate`) and then doing `pip install .`, which will get you all the right versions of the libraries needed. You'll need to do the `source` command there before running the script in a new terminal or anything.

(Docs for virtualenv are here: https://docs.python.org/3/library/venv.html)
